platform/core.multitabs/src/org/netbeans/core/multitabs/impl/TabDataRenderer.java

```diff
@@ -238,10 +238,10 @@ public void paint( Graphics g ) {
                 switch (tabsLocation) {
                     default:
                     case JTabbedPane.TOP:
-                        g.fillRect(0, rect.height - underlineHeight, rect.width, underlineHeight);
+                        g.fillRect(0, 0, rect.width, underlineHeight);
                         break;
                     case JTabbedPane.BOTTOM:
-                        g.fillRect(0, 0, rect.width, underlineHeight);
+                        g.fillRect(0, rect.height - underlineHeight, rect.width, underlineHeight);
                         break;
```

Comment:
@neilcsmith-net @eirikbakke  forgot to mention that I did swap the case ops as instructed.

It seems to have the desired effect.

Reply:
@mbien thanks, but thinking about this more, it would be better to swap the underline position based on the value of `EditorTab.underlineAtTop` as used at https://github.com/apache/netbeans/blob/master/platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/ui/FlatEditorTabCellRenderer.java#L68 ?  That should keep it consistent.

For same reason, I'd probably swap the horizontal tabs line position on the same flag?

Reply:
roger. I *think* i know what you mean. Going to take another look.

Reply:
implemented it finally. When the tab is on the bottom it is rendered on the bottom if `underlineAtTop` is set since everything is bottom up.

Reply:
I didn't touch left/right since the regular tabs don't change the line positioning there